File: src/go/day1/main.go
package main

import (
	"fmt"
	"slices"

	"github.com/dancantos/advent2024/src/go/pkg/timeit"
)

func main() {
	c1, c2 := slices.Clone(a1), slices.Clone(a2)

	fmt.Printf("Puzzle1 (sumdiffs): %d\n", sumdiffs(c1, c2))
	for range timeit.Run(100) {
		sumdiffs(c1, c2)
	}

	fmt.Println()

	c1, c2 = slices.Clone(a1), slices.Clone(a2)
	fmt.Printf("Puzzle2 (similarityScore): %d\n", similarityScore(c1, c2))
	for range timeit.Run(100) {
		similarityScore(c1, c2)
	}
}

File: src/python/day1/__main__.py
import timeit
from input import a1, a2
from puzzle1 import sumdiffs
from puzzle2 import similarityScore

def run():
    print(sumdiffs(a1, a2))
    print(similarityScore(a1, a2))
    diffsTime = timeit.timeit("sumdiffs(a1, a2)","from __main__ import sumdiffs, a1, a2", number=10)
    print("sumdiffs time: ", diffsTime/10*1000, "ms")
    simTime = timeit.timeit("similarityScore(a1, a2)","from __main__ import similarityScore, a1, a2", number=10)
    print("similarityScore time: ", simTime/10*1000, "ms")
    return

if __name__ == "__main__":
    run()